File: TelerikUIExamples/TelerikUIExamples/CustomCell.m
//
//  CustomCell.m
//  TelerikUIExamples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

#import "CustomCell.h"

static UIImage *gDayImage;
static UIImage *gSelectedDayImage;
static UIImage *gCurrentDayImage;

@interface CustomCell ()

@property (nonatomic, weak) UIImage *currentImg;

@end

@implementation CustomCell

+ (void)initialize
{
    gDayImage = [UIImage imageNamed:@"calendar_cell"];
    gDayImage = [gDayImage resizableImageWithCapInsets:UIEdgeInsetsMake(4, 4, 4, 4) resizingMode:UIImageResizingModeStretch];
    
    gCurrentDayImage = [UIImage imageNamed:@"calendar_current_cell"];
    gCurrentDayImage = [gCurrentDayImage resizableImageWithCapInsets:UIEdgeInsetsMake(4, 4, 4, 4) resizingMode:UIImageResizingModeStretch];
    
    gSelectedDayImage = [UIImage imageNamed:@"calendar_selected_cell"];
    gSelectedDayImage = [gSelectedDayImage resizableImageWithCapInsets:UIEdgeInsetsMake(12, 12, 12, 12) resizingMode:UIImageResizingModeStretch];
}

- (instancetype)initWithFrame:(CGRect)frame
{
    self = [super initWithFrame:frame];
    if (self) {
    }
    return self;
}

- (void)updateVisuals
{
    [super updateVisuals];
    
    self.style.backgroundColor = [UIColor clearColor];
    self.style.shape = nil;
    self.style.topBorderColor = nil;
    self.style.bottomBorderColor = nil;
    
    if (self.state & TKCalendarDayStateSelected) {
        self.currentImg = gSelectedDayImage;
    }
    else if (self.state & TKCalendarDayStateToday) {
        self.currentImg = gCurrentDayImage;
        self.label.textColor = [UIColor whiteColor];
    }
    else {
        self.currentImg = gDayImage;
    }
}

- (void)drawRect:(CGRect)rect
{
    [self.currentImg drawInRect:rect];
    [super drawRect:rect];
}

@end

File: TelerikUIExamplesInSwift2.0/TelerikUIExamplesInSwift/CustomCell.swift
//
//  CustomCell.swift
//  Swift Examples
//
//  Copyright (c) 2014 Telerik. All rights reserved.
//

import Foundation

struct CellImages
{
    static var dayImage: UIImage?
    static var currentDayImage: UIImage?
    static var selectedDayImage: UIImage?
 
    static func loadImages() {
        if (CellImages.dayImage == nil) {
            CellImages.dayImage = UIImage(named: "calendar_cell")
            CellImages.dayImage = CellImages.dayImage!.resizableImageWithCapInsets(UIEdgeInsetsMake(4, 4, 4, 4), resizingMode: UIImageResizingMode.Stretch)
        }
        
        if (CellImages.currentDayImage == nil) {
            CellImages.currentDayImage = UIImage(named: "calendar_current_cell")
            CellImages.currentDayImage = CellImages.currentDayImage!.resizableImageWithCapInsets(UIEdgeInsetsMake(4, 4, 4, 4), resizingMode: UIImageResizingMode.Stretch)
        }

        if (CellImages.selectedDayImage == nil) {
            CellImages.selectedDayImage = UIImage(named: "calendar_selected_cell")
            CellImages.selectedDayImage = CellImages.selectedDayImage!.resizableImageWithCapInsets(UIEdgeInsetsMake(12, 12, 12, 12), resizingMode: UIImageResizingMode.Stretch)
        }
    }
}

class CustomCell: TKCalendarDayCell {
    
    var currentImg: UIImage?
    
    override func updateVisuals() {
        super.updateVisuals()
    
        CellImages.loadImages()
        
        self.style()!.backgroundColor = UIColor.clearColor()
        self.style()!.shape = nil
        self.style()!.topBorderColor = nil
        self.style()!.bottomBorderColor = nil

        if self.state.rawValue & TKCalendarDayState.Selected.rawValue != 0 {
            self.currentImg = CellImages.selectedDayImage!
        }
        else if self.state.rawValue & TKCalendarDayState.Today.rawValue != 0 {
            self.currentImg = CellImages.currentDayImage!
            self.label.textColor = UIColor.whiteColor()
        }
        else {
            self.currentImg = CellImages.dayImage!
        }
    }
    
    override func drawRect(rect: CGRect) {
        if let img = self.currentImg {
            img.drawInRect(rect)
        }
        super.drawRect(rect)
    }
}